### PR TITLE
chore: migrate from memory-fs to Memfs

### PR DESCRIPTION
**Pull request by anikethsaha**
PR contains

- Migrate from `memory-fs` to `memfs`
- added lockfiles

### CODE REVIEW

**Comment by anikethsaha** [comment]
cc @evilebottnawi

**Comment by anikethsaha** [comment]
i think linter+prettier needs to be added in order to prevent this from happening in the future !

**Comment by evilebottnawi** [comment]
Yes, we need linter + prettier, but let's do it in a separate PR

**Comment by anikethsaha** [comment]
> Yes, we need linter + prettier, but let's do it in a separate PR

ok 👍

**Comment by anikethsaha** [comment]
cc @evilebottnawi

File: src/plugin.js [hunk 1]
```diff
@@ -98,7 +98,7 @@ NpmInstallPlugin.prototype.preCompile = function(compilation, next) {
     );
 
     this.preCompiler = webpack(config);
-    this.preCompiler.outputFileSystem = new MemoryFS();
+    this.preCompiler.outputFileSystem = createFsFromVolume(new Volume());
```
Comment (evilebottnawi) [inline src/plugin.js]: Please add `this.preCompiler.outputFileSystem.join = path.join.bind(path)`, we need support webpack@4

**Comment by anikethsaha** [comment]
cc @evilebottnawi

**Pull request merged**